src/main/cpp/src/get_json_object.hpp

```diff
@@ -19,10 +19,596 @@
 #include <cudf/strings/string_view.hpp>
 #include <cudf/strings/strings_column_view.hpp>
 
+#include <thrust/optional.h>
+#include <thrust/pair.h>
+#include <thrust/scan.h>
+#include <thrust/tuple.h>
+
 #include <memory>
 
 namespace spark_rapids_jni {
 
+namespace detail {
```

Comment:
So everything in this file and `json_parser.hpp` are running on host CPU?

Reply:
In production, `json_parser.hpp`  runs on GPU.
But for the unit test cases in `json_parser_tests.cpp`, they are running on CPU.

`json_parser` `json_start_pos` scans the JSON string view on GPU and writes output to cuDF string column.

`json_parser` is straightforward, it scans char array(parameter is `char *`) and writes to char buffer (parameter is `char *`). So it's convenient to use `CUDF_HOST_DEVICE` to test the logic.
Sorry for introduced extra `HOST` annotation.

Reply:
Although the code can run on both CPU and GPU, I suspect that there may be some parts that cannot run exactly the same way on both places. We should better to implement it only as GPU code, and run unit tests on the GPU data.

Reply:
OK, will update